src/use-dispatch/use-dispatch.ts

```diff
@@ -49,6 +49,7 @@ export class UseDispatch extends StimulusUse {
     // dispatch the event from the given element or by default from the root element of the controller
     targetElement.dispatchEvent(event)
 
+    warn('dispatch', 'useDispatch() is a deprecated mixin from Stimulus-Use. Please use the built-in dispatch() function from Stimulus: https://stimulus.hotwired.dev/reference/controllers#cross-controller-coordination-with-events');
```

Comment:
This doesn't seem to work. 

Ironically if you have this controller here:

```js
import { Controller } from '@hotwired/stimulus'
import { useDispatch } from 'stimulus-use'

export default class extends Controller {
  connect() {
    useDispatch(this)
  }

  add() {
    this.dispatch('add', { quantity: 1 })
  }
}
```
It's already using the `this.dispatch()` function from Stimulus itself. So this code in our `dispatch` function is actually never called again. 

Which is why we should move the deprecation warning into our `useDispatch()` function itself.

Reply:
Hm, that's odd...
When I run your controller example with two additional console.log outputs:

```js
import { Controller } from '@hotwired/stimulus'
import { useDispatch } from 'stimulus-use'

export default class extends Controller {
  connect() {
    console.log("ITEM Controller connected");
    useDispatch(this)
  }

  add() {
    console.log("ITEM Controller#add was called");
    this.dispatch('add', { quantity: 1 })
  }
}
```

I get following Output in the Console:

![image](https://user-images.githubusercontent.com/17851143/197958871-7b2dc8e3-638c-4f61-86a5-91239d29c837.png)


---

The Reason why I didn't put the deprecation warning into the `useDispatch()` function itself is, it would print the warning on every request, even though I don't use or import the `useDispatch()` function.
That's why I put the deprecation warning in the `dispatch()` call.

Do you thinks that's a bad idea?